commit/plugin.go

```diff
@@ -250,66 +254,78 @@ func (p *Plugin) Observation(
 		return nil, fmt.Errorf("decode query: %w", err)
 	}
 
+	lggr = logger.With(
+		lggr,
+		"prevOutcome", prevOutcome,
+		"decodedQuery", decodedQ,
```

Comment:
I think we should make sure these are logged once at the start of our top level observation functions in an info/verbose log rather than in (potentially) multiple log statements. The outcome could be huge.

Reply:
Will axe for now.